src/main/java/org/apache/commons/lang3/builder/HashCodeBuilder.java

```diff
@@ -180,6 +181,10 @@ private static void reflectionAppend(final Object object, final Class<?> clazz,
         if (isRegistered(object)) {
             return;
         }
+        if (object instanceof CharSequence || object instanceof Number || object instanceof Temporal) {
```

Comment:
Needs a // comment IMO. Why not refactor the test exporession into a common (if possible package-private) method with the same code above?